### DISCUSSION TITLE
Add-PnPPage: Does not create site page URL with spaces

### DISCUSSION THREAD

**Discussion by AJD8** [pnp/powershell#511] [Q&A]
I have changed over from SharePointPnPPowerShellOnline to PnP.PowerShell v1.4 and it appears I am no longer able to create a site page that has a URL with spaces.

Example:
$CompanyName = 'test page'
Add-PnPPage -Name $CompanyName -CommentsEnabled:$false
Set-PnPPage -Identity $CompanyName -HeaderType None

What will happen is the site page will be created with a URL such as test-page or test+page.  Due to the addition of the '-' or '+' the set page will fail as the company name no longer matches.

Is this expected behavior?  I also tried adding '%20' for the space on the add-pnppage, but that leads to a "Object reference not set to an instance of an object." error.

Additonal info, I am running PnP.Powershell in Powershell v5.1.

**Comment by BaronSparky** [comment]
I think the page name in your example, would be "`test page.aspx`", hence why the `Set-PnPPage` fails perhaps?  (There is no page called "test page")

Have you tried something like:
```
$CompanyName = 'test page'
$page = Add-PnPPage -Name $CompanyName -CommentsEnabled:$false
Set-PnPPage -Identity $page.Name -HeaderType None
```

**Reply by AJD8** [reply]
Thanks.  I may have not phrased my Q well, but I want the page to accept the spaces.  But if I try passing the "test page" in as the name it fills the space with a '-' or a'+' which is not the desired result.

**Comment by BaronSparky** [comment]
I think it does though what you are asking for:
![image](https://user-images.githubusercontent.com/11838512/113102477-ffd38780-91f5-11eb-98ed-5a7dbcc3278a.png)

It isn't adding anything to the page name or title. When you open the page, the URL cannot contain a space hence why it is escaped with "%20".

You can also create the page with a %20 - "test%20page" works in my environment.

**Reply by AJD8** [reply]
This is what I am getting when I try same thing:
![image](https://user-images.githubusercontent.com/81631053/113139485-3066f180-91f5-11eb-987e-7414e050e992.png)


I passed in these commands as part of a larger script:

$page = Add-PnPPage -Name $CompanyName -CommentsEnabled:$false -ErrorAction SilentlyContinue
    echo "this is the Page Name: " $page.Name
    Set-PnPPage -Identity $page.Name -HeaderType None
	Write-Host ""

**Reply by AJD8** [reply]
I was able to confirm behavior with a consulting group we work with, I am going to submit a bug.
